plugins/ai/copilot-chat.nix

```diff
@@ -0,0 +1,316 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "copilot-chat";
+  originalName = "CopilotChat.nvim";
+  defaultPackage = pkgs.vimPlugins.CopilotChat-nvim;
+
+  maintainers = [ maintainers.GaetanLepage ];
+
+  settingsOptions = {
+    debug = helpers.defaultNullOpts.mkBool false ''
+      Enable debug logging.
+    '';
+
+    proxy = helpers.defaultNullOpts.mkStr null ''
+      Custom proxy to use: `[protocol://]host[:port]`.
+    '';
+
+    allow_insecure = helpers.defaultNullOpts.mkBool false ''
+      Allow insecure server connections.
+    '';
+
+    system_prompt = helpers.defaultNullOpts.mkLua "require('CopilotChat.prompts').COPILOT_INSTRUCTIONS" ''
+      System prompt to use.
+    '';
+
+    model = helpers.defaultNullOpts.mkStr "gpt-4" ''
+      GPT model to use, 'gpt-3.5-turbo' or 'gpt-4'.
+    '';
+
+    temperature = helpers.defaultNullOpts.mkNullable (types.number.between 0.0 1.0) 0.1 ''
+      GPT temperature.
+    '';
+
+    question_header = helpers.defaultNullOpts.mkStr "## User " ''
+      Header to use for user questions.
+    '';
+
+    answer_header = helpers.defaultNullOpts.mkStr "## Copilot " ''
+      Header to use for AI answers.
+    '';
+
+    error_header = helpers.defaultNullOpts.mkStr "## Error " ''
+      Header to use for errors.
+    '';
+
+    separator = helpers.defaultNullOpts.mkStr "───" ''
+      Separator to use in chat.
+    '';
+
+    show_folds = helpers.defaultNullOpts.mkBool true ''
+      Shows folds for sections in chat.
+    '';
+
+    show_help = helpers.defaultNullOpts.mkBool true ''
+      Shows help message as virtual lines when waiting for user input.
+    '';
+
+    auto_follow_cursor = helpers.defaultNullOpts.mkBool true ''
+      Auto-follow cursor in chat.
+    '';
+
+    auto_insert_mode = helpers.defaultNullOpts.mkBool false ''
+      Automatically enter insert mode when opening window and if auto follow cursor is enabled on new prompt.
+    '';
+
+    clear_chat_on_new_prompt = helpers.defaultNullOpts.mkBool false ''
+      Clears chat on every new prompt.
+    '';
+
+    highlight_selection = helpers.defaultNullOpts.mkBool true ''
+      Highlight selection.
+    '';
+
+    context =
+      helpers.defaultNullOpts.mkEnum
+        [
+          "buffers"
+          "buffer"
+        ]
+        null
+        ''
+          Default context to use, `"buffers"`, `"buffer"` or `null` (can be specified manually in prompt via @).
+        '';
+
+    history_path = helpers.defaultNullOpts.mkStr (helpers.mkRaw "vim.fn.stdpath('data') .. '/copilotchat_history'") ''
```

Comment:
Passing a raw type here is tripping `mkStr`'s assertion. It's the assertion that's wrong though.

It should be `assert default == null || isSteing default || isRaw default`. But realistically, we can remove the assertion now that we know `mkDesc` & `toPretty` are working correctly.